api/grpc_server.go

```diff
@@ -71,24 +71,26 @@ func (s SpacemeshGrpcService) GetNonce(ctx context.Context, in *pb.AccountId) (*
 func (s SpacemeshGrpcService) SubmitTransaction(ctx context.Context, in *pb.SignedTransaction) (*pb.TxConfirmation, error) {
 	log.Info("GRPC SubmitTransaction msg")
 
-	signedTx := &types.SerializableSignedTransaction{}
-	err := types.BytesToInterface(in.Tx, signedTx)
+	tx, err := types.BytesAsTransaction(in.Tx)
 	if err != nil {
 		log.Error("failed to deserialize tx, error %v", err)
 		return nil, err
 	}
-	log.Info("GRPC SubmitTransaction to address: %s (len: %v), amount: %v gaslimit: %v, price: %v", signedTx.Recipient.Short(), len(signedTx.Recipient), signedTx.Amount, signedTx.GasLimit, signedTx.GasPrice)
-	_, err = s.Tx.ValidateTransactionSignature(signedTx)
-	if err != nil {
-		log.Error("tx failed to validate signature, error %v", err)
+	log.Info("GRPC SubmitTransaction to address: %s (len: %v), amount: %v gaslimit: %v, price: %v",
+		tx.Recipient.Short(), len(tx.Recipient), tx.Amount, tx.GasLimit, tx.GasPrice)
+	if err := tx.CalcAndSetOrigin(); err != nil {
+		log.With().Error("failed to calc origin", log.Err(err))
+		return nil, err
+	}
+	if !s.Tx.AddressExists(tx.Origin()) {
```

Comment:
shouldn't you check tx validity here? this will allow the user to get faster validation for newly created tx

Reply:
Yes, it's a little tricky to get the required state here, so I'd rather postpone this to after this PR is merged.

Reply:
Please add a technical debt and comment. I think this is crucial to when we roll out testnet

Reply:
I'll create an issue for this